test/Microsoft.AspNetCore.Mvc.Formatters.Json.Test/JsonInputFormatterTest.cs

```diff
@@ -571,7 +571,7 @@ public void CreateJsonSerializer_UsesJsonSerializerSettings()
         [InlineData("{\"a\":{\"b\"}}", "a", "Invalid character after parsing property name. Expected ':' but got: }. Path 'a', line 1, position 9.")]
         [InlineData("{\"age\":\"x\"}", "age", "Could not convert string to decimal: x. Path 'age', line 1, position 10.")]
         [InlineData("{\"login\":1}", "login", "Error converting value 1 to type 'Microsoft.AspNetCore.Mvc.Formatters.JsonInputFormatterTest+UserLogin'. Path 'login', line 1, position 10.")]
-        [InlineData("{\"login\":{\"username\":\"somevalue\"}}", "login", "Required property 'Password' not found in JSON. Path 'login', line 1, position 33.")]
+        [InlineData("{\"login\":{\"username\":\"somevalue\"}}", "login.Password", "Required property 'Password' not found in JSON. Path 'login', line 1, position 33.")]
```

Comment:
But doesn't this approach fail if the child and parent property names are the same?

e.g. JSON: `"{\"login\":{\"username\":\"somevalue\"}}"`, and the required member is also `login`? The path will be `login`, not `login.login`.

Reply:
How do you mean "fail"? When the parent and child have the same name, there's no way for the error handler to know whether `login` or `login.login` are correct. It chooses `login` but that's an arbitrary choice.

Reply:
If the conditions can be simplified more than our offline discussions led me to believe, please let me know. Would be wonderful if we can assume `Member` is *not* included in `Path`.

Reply:
> How do you mean "fail"?

This is producing inconsistent paths. In the case of a missing required property throwing an exception, sometimes the path points to the missing member, and other times it points to the parent object. Anyone relying using path programmatically and expects it to behave in a consistent way will eventually stumble upon this edge case. What do they do at that point?

In the original issue the author says "the resulting error do[es] not show the field name" but the exception message for a missing required property does include the field name:
 `Required property 'Password' not found in JSON. Path 'login', line 1, position 33.`. Why isn't he using the exception message?

Reply:
> This is producing inconsistent paths.

This is improving on the inconsistent paths today. Json.NET does not consistently include `Member` in `Path` and we need to add it when we can be sure it's missing.

> Why isn't he using the exception message?

Nobody parses exception messages.

Reply:
I think the way to handle this properly would be to add a ErrorType to ErrorContext in Newtonsoft.Json. Using that you could test that the error was thrown because of a missing required property and know that the JSON path is the object and not the property.

Reply:
That would make things even more consistent but couldn't go in now. I suggest we get this in now and do what you suggest if / when we upgrade our NewtonSoft.Json dependency to a version that includes your proposed property.

Reply:
I think it is better to wait for a consistent fix.

Also the path is used to build up the model state. Changing the key and the resolved `ModelMetadata` for an exception is changing MVC's existing behavior. In the case of an exception from a missing required property this change might be desirable, but this code runs for all serialization errors, and for some of them it might make sense to not be included. I think a change like this needs time in dev and preview for us/customers to find potential regressions.

Reply:
What I'm putting in is what we agreed on as a fix for #8509 some time ago. We're highly unlikely to get a consistent fix because we're not going to upgrade to the latest NewtonSoft.Json anytime soon. So, you're basically saying we shouldn't fix this bug and I have trouble with that.

Note this is a corner case and ending up with a still-inconsistent path is an _extreme_ corner case. For example, I only updated a couple of existing tests.

@mkArtakMSFT what do you think about never fixing #8509?